### PR TITLE
Fixing audit log for companies

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | 

#### Description:

We were missing the owner changes in the company audit log. This PR fixes that.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Change company owner
3. Check the audit_log table that the owner change is present there.

### CODE REVIEW

**Review by dadarya0 — DISMISSED** [review]
Thanks @escopecz

File: app/bundles/LeadBundle/Entity/Company.php [hunk 1]
```diff
@@ -178,6 +178,33 @@ public static function getDefaultIdentifierFields(): array
         ];
     }
 
+    protected function isChanged($prop, $val)
+    {
+        $prefix = 'company';
+
+        if (substr($prop, 0, strlen($prefix)) === $prefix) {
```
Comment (dadarya0) [inline app/bundles/LeadBundle/Entity/Company.php]:
```suggestion
        if (str_starts_with($prop, $prefix)) {
```

Comment (dadarya0) [reply app/bundles/LeadBundle/Entity/Company.php]: same is failing on rector.

**Review by dadarya0 — APPROVED** [review]

**Review by rohitpavaskar — APPROVED** [review]
Tested, Working as expected.

**Pull request merged**